critiquebrainz/frontend/external/spotify_test.py

```diff
@@ -63,76 +61,86 @@ def setUp(self):
         }
         self.fresh_token_response = MockResponse(content=self.fresh_token)
 
-    def test_get(self):
+    @mock.patch('critiquebrainz.frontend.external.spotify._fetch_access_token')
+    @mock.patch('requests.get')
+    def test_valid_get(self, requests_get, fetch_spotify_token):
         # test when _fetch_access_token returns acceptable token
-        requests.get = MagicMock(return_value=self.ok_response)
-        spotify._fetch_access_token = MagicMock(return_value="valid-token")
+        requests_get.return_value = self.ok_response
+        fetch_spotify_token.return_value = "valid-token"
         self.assertDictEqual(spotify._get("test-query"), self.some_albums_response)
-        spotify._fetch_access_token.assert_called_with(refresh=False)
+        fetch_spotify_token.assert_called_with(refresh=False)
 
         # test when _fetch_access_token returns expired token
-        requests.get = MagicMock(return_value=self.unauth_response)
-        spotify._fetch_access_token = MagicMock(return_value="expired-token")
+        requests_get.return_value = self.unauth_response
```

Comment:
if we use the mock twice in a single test, we should use `.reset_mock()` on it before setting new values and asserting that it is called